google-cloud-storage/src/main/java/com/google/cloud/storage/BlobId.java

```diff
@@ -114,6 +120,21 @@ public static BlobId of(String bucket, String name, Long generation) {
     return new BlobId(checkNotNull(bucket), checkNotNull(name), generation);
   }
 
+  /**
+   * Creates a {@code BlobId} object.
+   *
+   * @param storageUrl the Storage url to create the blob from
+   */
+  public static BlobId fromStorageUrl(String storageUrl) {
+    if(!Pattern.matches("gs://.*/.*", storageUrl)) {
+      throw new IllegalArgumentException(storageUrl + " is not a valid Storage URL");
+    }
+    String bucketName = storageUrl.split("/")[2];
+    String blobName = storageUrl.split(bucketName + "/")[1];
```

Comment:
Possibly simplify string allocation and traversal?

```suggestion
    int blobNameStartIndex = storageUrl.indexOf('/', 5);
    String bucketName = storageUrl.substring(5, blobNameStartIndex);
    String blobName = storageUrl.substring(blobNameStartIndex);
```